src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1483,37 +1485,45 @@ public void setAllowFlight(boolean flight)
 	@Deprecated
 	public void hidePlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		createRecordOfHiddenPlayerIfNotExists(player);
 	}
 
 	@Override
 	public void hidePlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		createRecordOfHiddenPlayerIfNotExists(player);
+		hiddenPlayerNicks.get(player.getUniqueId()).add(plugin.getName());
 	}
 
 	@Override
 	@Deprecated
 	public void showPlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hiddenPlayerNicks.remove(player.getUniqueId());
 	}
 
 	@Override
 	public void showPlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(hiddenPlayerNicks.containsKey(player.getUniqueId()))
+		{
+			List<String> blockingPlugins = hiddenPlayerNicks.get(player.getUniqueId());
+			blockingPlugins.remove(plugin.getName());
+			if(blockingPlugins.size() == 0)
```

Comment:
`.size() == 0` can be simplified.
```suggestion
			if (blockingPlugins.size().isEmpty())
```

Reply:
Surely not `.size().isEmpty()`?
```suggestion
			if(blockingPlugins.empty())
```